torchdata/datapipes/iter/util/cacheholder.py

```diff
@@ -305,12 +312,63 @@ def _wait_promise_fn(timeout, filename):
     return filename
 
 
-class _FulfilledPromisesIterDataPipe(IterDataPipe):
+@functional_datapipe("memory_cell")
+class _MemoryCellIterDataPipe(IterDataPipe):
+    def __init__(self, source_datapipe, remember_elements=10):
+        self.source_datapipe = source_datapipe
+        self.buffer = [None for i in range(remember_elements)]
+        self.remember_elements = remember_elements
```

Comment:
We do need to remember more than one in case of empty output filter. Imagine situation like: 
```
rec1 -> []
rec2-> []
rec3->[new_rec1, new_rec2]
```
We will get control only at `new_rec1`, meaning somehow we need access to `rec1`,`rec2`